pysatModels/tests/test_models.py

```diff
@@ -70,3 +71,64 @@ def teardown_class(self):
         self.tempdir.cleanup()
         del self.inst_loc, self.saved_path, self.tempdir
         return
+
+
+class TestSAMIPysatVersion(object):
+    """Test SAMI load code for pysat version differences across v3.0.2."""
+
+    def setup_class(self):
+        """Initialize the testing setup once before all tests are run."""
+        # Make sure to use a temporary directory so that the user setup is not
+        # altered
+        self.tempdir = tempfile.TemporaryDirectory()
+        self.saved_path = pysat.params['data_dirs']
+        self.saved_ver = pysat.__version__
+        pysat.params.data['data_dirs'] = [self.tempdir.name]
+
+        # Assign the location of the model Instrument sub-modules
+        self.inst_loc = pysatModels.models
+        return
+
+    def teardown_class(self):
+        """Clean up downloaded files and parameters from tests."""
+
+        pysat.params.data['data_dirs'] = self.saved_path
+        pysat.__version__ = self.saved_ver
+
+        self.tempdir.cleanup()
+        del self.inst_loc, self.saved_path, self.tempdir, self.saved_ver
+        return
+
+    def test_against_version(self):
+        """Test for SAMI load failure for pysat version opposite v3.0.2."""
```

Comment:
'opposite' doesn't make sense in this sentence.  Separately, shouldn't this be something like `def test_load_failure` and a similar description?  The same test is done, just has a version loop.  Also, there should be a TODO comment+issue for when the logic can be removed.

Reply:
How long do you want compatibility with pysat v3.0.1?

Reply:
Another thing we should probably discuss.  But it can be an issue and settled at our next meeting, allowing this PR to move forward.